packages/devtools_app/lib/src/config_specific/drag_and_drop/drag_and_drop.dart

```diff
@@ -20,13 +18,13 @@ abstract class DragAndDropManager {
     init();
   }
 
-  static DragAndDropManager get instance => _instance ?? DragAndDropManager();
+  static DragAndDropManager get instance => _instance;
 
-  static DragAndDropManager _instance;
+  static late DragAndDropManager _instance = DragAndDropManager();
```

Comment:
instead of making this late and assigning _instance to `DragAndDropManager()`, what if we do this:
```
static DragAndDropManager get instance => _instance != null ? instance! : DragAndDropManager();
static late final DragAndDropManager? _instance;
```

This is more of an equivalent to the code we had before, where _instance is only being assigned once.

Reply:
late is a right way to implement lazy pattern: https://dev.to/pktintali/late-variables-in-dart-dart-learning-series-1-2opf
For some reasons `init` assigns it to `this`. I do not understand it so I left it as is and that's why the variable is not final.

But the code before initialized the field more than once.

What do you think?

Reply:
In the previous code, the only place _instance is assigned is in init: `_instance = this;`. This code is a bit confusing though because `_instance = DragAndDropManager()` is essentially the same thing as creating the DragAndDropManager constructor which calls init and sets `_instance` to `this` anyway.

How about this:
```
static DragAndDropManager get instance => _instance ??= DragAndDropManager(); 
static late final DragAndDropManager _instance;
```

And by doing this, you can get rid of the init method body all together. Make init an abstract method `void init();` so that subclasses still have to override it (you can add an empty override for the desktop impl). You can get rid of the mustCallSuper annotation then too.

Reply:
done